File: 백준/Gold/1027. 고층 건물/고층 건물.cs
int n = int.Parse(Console.ReadLine()!);
string[] inputs = Console.ReadLine()!.Split();

var towers = new double[n];
for (int i = 0; i < n; i++)
{
    towers[i] = double.Parse(inputs[i]);
}

int max = 0;
for (int i = 0; i < n; i++)
{
    int cnt = 0;
    double diff = -1_000_000_000;
    for (int j = 1; j <= i; j++)
    {
        if (towers[i - j] - towers[i] > diff * j)
        {
            cnt++;
            diff = (towers[i - j] - towers[i]) / j;
        }
    }

    diff = -1_000_000_000;
    for (int j = 1; j < n - i; j++)
    {
        if (towers[i + j] - towers[i] > diff * j)
        {
            cnt++;
            diff = (towers[i + j] - towers[i]) / j;
        }
    }

    if (max < cnt)
    {
        max = cnt;
    }
}

Console.WriteLine(max);

File: 백준/Gold/1027. 고층 건물/고층 건물.rs
use std::io::{stdin, Read};

fn main() {
    let mut buf = String::new();
    stdin().read_to_string(&mut buf).expect("input error");
    let mut input = buf.split_whitespace().flat_map(str::parse::<f64>);

    let n = input.next().unwrap() as usize;
    let mut tower = vec![0_f64; n];
    for i in 0..n {
        tower[i] = input.next().unwrap();
    }

    let mut max = 0;
    for i in 0..n {
        let mut cnt = 0;
        let mut diff = -1000000000_f64;
        for j in 1..=i {
            if tower[i - j] - tower[i] > diff * j as f64{
                cnt += 1;
                diff = (tower[i - j] - tower[i]) / j as f64;
            }
        }
        diff = -1000000000_f64;
        for j in 1..(n - i) {
            if tower[i + j] - tower[i] > diff * j as f64{
                cnt += 1;
                diff = (tower[i + j] - tower[i]) / j as f64;
            }
        }

        if max < cnt {
            max = cnt;
        }
    }

    println!("{max}");
}
